examples/codegen/README.md

```diff
@@ -14,17 +14,74 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-Example of how to run the pipeline:
+## ONNX Export
+Firstly, we need to install HuggingFace optimum library
+```bash
+pip install optimum
+```
+
+### Patch the original PyTorch Model
+TODO
```

Comment:
```suggestion
First apply the following modification to this file in your transformers installation:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/codegen/modeling_codegen.py#L212

\```diff
-offset = layer_past[0].shape[-2]
+offset = (attention_mask[0] == 0.0).sum() - 1.0
\```

We need to do this because the existing with_past implementations assume there is no padding in the inputs. With deepsparse, we need to use static sequence length, which means our offset for the embeddings will depend on how many non-padded inputs we receive.

The new line checks this with the attention_mask. At this point in the code, attention_mask has been transformed from a tensor with 0s and 1s, to a tensor of `float.min` and `0.0`. So when we compare `attention_mask == 0.0` we are actually saying everywhere the attention_mask is 1.

We also need to subtract 1 from this count, because the attention mask is applied AFTER the kv cache is concatenated to the new token, which means the attention mask will actually be sequence length + 1 items. So we subtract 1 to get the current sequence length.
```